File: src/resources/api-keys.ts
import type { HttpClient } from '../http.js';
import type { ApiKeyRecord, CreateApiKeyBody } from '../types/misc.js';

export class ApiKeysResource {
  constructor(private http: HttpClient) {}

  async create(body?: CreateApiKeyBody): Promise<ApiKeyRecord> {
    return this.http.post<ApiKeyRecord>('/v1/api-keys', body ?? {});
  }

  async list(): Promise<ApiKeyRecord[]> {
    return this.http.get<ApiKeyRecord[]>('/v1/api-keys');
  }

  async revoke(id: string): Promise<{ revoked: boolean }> {
    return this.http.delete<{ revoked: boolean }>(`/v1/api-keys/${id}`);
  }
}

File: packages/python/invariance/resources/api_keys.py
from __future__ import annotations

from typing import Any

from ..http_client import HttpClient
from ..types import ApiKeyRecord, CreateApiKeyBody


class ApiKeysResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def create(self, body: CreateApiKeyBody | None = None) -> ApiKeyRecord:
        return await self._http.post("/v1/api-keys", body or {})

    async def list(self) -> list[ApiKeyRecord]:
        return await self._http.get("/v1/api-keys")

    async def revoke(self, id: str) -> dict[str, bool]:
        return await self._http.delete(f"/v1/api-keys/{id}")
